server/app/com/xsn/explorer/tasks/MasternodeSynchronizerTask.scala

```diff
@@ -0,0 +1,72 @@
+package com.xsn.explorer.tasks
+
+import akka.actor.{Actor, ActorSystem, Props}
+import com.xsn.explorer.config.MasternodeSynchronizerConfig
+import com.xsn.explorer.models._
+import com.xsn.explorer.services.{XSNService}
+import javax.inject.Inject
+import org.scalactic.{Bad, Good}
+import org.slf4j.LoggerFactory
+import com.xsn.explorer.models.values.IPAddress
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+class MasternodeSynchronizerActor extends Actor {
```

Comment:
While this is a nice usage for the actors, it's usually not good to expose such interfaces, instead, I would create something like:


```scala
trait MasternodeService {
  def get(...): Future[List[Masternode]]
}

object MasternodeService {
  class ActorImpl @Inject()(actor: MasternodeSynchronizerActor.Ref) extends MasternodeService {
    override def get(...): Future[List[Masternode]] = {
      actor.ask(...)...
    }
  }
}
```

A missing piece would be like:

```scala
object MasternodeSynchronizerActor {
  final class Ref private (val ref: ActorRef) extends AnyVal

  object Ref {

    def apply(name: String = "masternode-synchronizer")(implicit ec: ExecutionContext, system: ActorSystem): Ref = {
      val actor = system.actorOf(Props(new MasternodeSynchronizerActor), name)
      new Ref(actor)
    }
  }
}

```